language/stdlib/modules/libra_system.mvir

```diff
@@ -24,7 +24,18 @@ module LibraSystem {
 
     resource BlockMetadata {
       // Height of the current block
+      // TODO: Should we keep the height?
       height: u64,
+
+      // current real world time provided by consensus.
+      timestamp: u64,
+
+      // hash of the previous block
+      id: bytearray,
+
+      // A serialized btreemap of block votes from previous block.
```

Comment:
- I thought our plan was to emit this data in an event for each block so consensus can use it, but not store it on-chain. Is there a use-case for also storing it on-chain (cc @aching)
- Not necessarily related, but: why is this a btreemap instead of an array? That seems overspecified + will include structural overhead that we probably don't need.